Add cloudformation support for creating lambda profiling groups

pandpara opened the pull request:
*Description of changes:*

As CodeGuru Profiler has now support for lambda profiling groups, we would like to also cover that in Cloudformation. This change allows user to specify compute platform["AWSLambda"/"Default"] for ProfilingGroup at creation.

CreateProfilingGroup API with compute platform : https://docs.aws.amazon.com/codeguru/latest/profiler-api/API_CreateProfilingGroup.html#profiler-CreateProfilingGroup-request-computePlatform

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

mirelap-amazon commented on aws-codeguruprofiler-profilinggroup/src/main/java/software/amazon/codeguruprofiler/profilinggroup/CreateHandler.java:
```diff
@@ -40,13 +40,18 @@
-            .build();
+            .clientToken(request.getClientRequestToken());
+
+        if (computePlatform != null) {
+            createProfilingGroupRequestBuilder.computePlatform(computePlatform);
+        }
```
What happens if you pass *null* to the request? If nothing changes you could pass directly the value.

Or consider initializing with the default first and then use directly the *build()* from the *Builder*.

```
String computePlatform = (model.getComputePlatform() == null) ? "Default" : model.getComputePlatform();

CreateProfilingGroupRequest createProfilingGroupRequest = ... 
    .computePlatform(computePlatform)
    .build();
```

pandpara replied:
SDK handles `null` case for non-required fields, so we don't need to have a null check here. Removed.

> 1. I've left some minor comments, change looks good.
> 2. Please take a look at the failed build from CodeBuild, it looks like _aws-cloudformation-rpdk-java-plugin_.

For point 2: I have fixed it. The issue was because of older _aws-cloudformation-rpdk-java-plugin_ version. Updated to `2.0.0` as suggested.

pandpara replied:
> What happens if you pass _null_ to the request? If nothing changes you could pass directly the value.
> 
> Or consider initializing with the default first and then use directly the _build()_ from the _Builder_.
> 
> ```
> String computePlatform = (model.getComputePlatform() == null) ? "Default" : model.getComputePlatform();
> 
> CreateProfilingGroupRequest createProfilingGroupRequest = ... 
>     .computePlatform(computePlatform)
>     .build();
> ```

SDK handles null case for non-required fields, so we don't need to have a null check here. Removed.

pandpara commented on aws-codeguruprofiler-profilinggroup/aws-codeguruprofiler-profilinggroup.json:
```diff
@@ -20,6 +20,14 @@
+      "description": "The compute platform of the profiling group.",
+      "type": "string",
+      "enum": [
+        "Default",
```
As mentioned in the CreateProfilingGroup request doc : https://docs.aws.amazon.com/codeguru/latest/profiler-api/API_CreateProfilingGroup.html#profiler-CreateProfilingGroup-request-computePlatform

Valid values for `ComputePlatform` are :
* Default
* AWSLambda

And hence when we create a profiling group via CLI it would accept these two values for the _ComputePlatform_. Similarly, the CloudFormation should support these valid values.

Whereas having _Others_ on the console is a UX choice decided during the review.

pandpara replied:
> It looks like the UI offers the _Lambda_ and _Other_ options for the compute platform, while the CloudFormation offers _Lambda_ and _Default_:
> 
> * https://docs.aws.amazon.com/codeguru/latest/profiler-ug/setting-up.html#setting-up-step-2
> * https://docs.aws.amazon.com/codeguru/latest/profiler-api/API_CreateProfilingGroup.html#profiler-CreateProfilingGroup-request-computePlatform
> 
> As part of this PR, it looks we'll keep the _Default_ to be consistent, but maybe we should consider updating the UI to be consistent.

As mentioned in the CreateProfilingGroup request doc : https://docs.aws.amazon.com/codeguru/latest/profiler-api/API_CreateProfilingGroup.html#profiler-CreateProfilingGroup-request-computePlatform

Valid values for `ComputePlatform` are :
* Default
* AWSLambda

And hence when we create a profiling group via CLI it would accept these two values for the _ComputePlatform_. Similarly, the CloudFormation should support these valid values.

Whereas having _Others_ on the console is a UX choice decided during the review.

pandpara commented on aws-codeguruprofiler-profilinggroup/sample-template.json:
```diff
@@ -8,6 +8,20 @@
         "ProfilingGroupName": "MySampleProfilingGroup"
       }
     },
+    "MyProfilingGroupWithDefaultComputePlatform": {
+      "Type": "AWS::CodeGuruProfiler::ProfilingGroup",
+      "Properties": {
+        "ProfilingGroupName": "MyProfilingGroupWithDefaultComputePlatform",
+        "ComputePlatform": "Default"
+      }
+    },
+    "MyProfilingGroupWithLambdaComputePlatform": {
+      "Type": "AWS::CodeGuruProfiler::ProfilingGroup",
+      "Properties": {
+        "ProfilingGroupName": "MyProfilingGroupWithLambdaComputePlatform",
+        "ComputePlatform": "AWSLambda"
+      }
+    },
```
> For backwards compatibility reasons, add one without any _ComputePlatform_ property set to test it and have it as a sample

There's already a sample profiling group without any _ComputePlatform_ at Line 5:10

ivoanjo commented:
Hey @pandpara , re: the broken build, we were talking about this on our chat ([opaque Amazon-only link](https://tiny.amazon.com/1etbaav2l)) on how we can look into the CI issues. It looks like with the current setup, only Amazon employees can look at the setup, but it would be nice for us to document that in the `README` as I suggested in the chat thread linked. Could you look into it?

pandpara commented:
> Hey @pandpara , re: the broken build, we were talking about this on our chat ([opaque Amazon-only link](https://tiny.amazon.com/1etbaav2l)) on how we can look into the CI issues. It looks like with the current setup, only Amazon employees can look at the setup, but it would be nice for us to document that in the `README` as I suggested in the chat thread linked. Could you look into it?

I have updated the `README` with suggested note around CI failures.

pandpara commented:
> 1. I've left some minor comments, change looks good.
> 2. Please take a look at the failed build from CodeBuild, it looks like _aws-cloudformation-rpdk-java-plugin_.

pandpara commented:
> 1. I've left some minor comments, change looks good.
> 2. Please take a look at the failed build from CodeBuild, it looks like _aws-cloudformation-rpdk-java-plugin_.

For point 2: I have fixed it. The issue was because of older _aws-cloudformation-rpdk-java-plugin_ version. Updated to `2.0.0` as suggested.

pandpara commented on aws-codeguruprofiler-profilinggroup/README.md:
```diff
@@ -68,3 +68,7 @@ pre-commit run --all-files && AWS_REGION=us-east-1 mvn clean verify package
+
+#### Dealing with CI Failures
+If outside of Amazon, you will not be able to see the CI outputs. Our reviewers will be responsible for making that
+information available to you. Reviewers, you can find info on how to get CI outputs at [link](https://tinyurl.com/yagdpv7q).
```
Fixed! :)
